src/sonic-yang-models/yang-models/sonic-dhcp-server-ipv4.yang

```diff
@@ -0,0 +1,237 @@
+module sonic-dhcp-server-ipv4 {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/sonic-dhcp-server-ipv4";
+
+    prefix dhcp-server-ipv4;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-vlan {
+        prefix vlan;
+    }
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+    }
+
+    description "DHCP_SERVER_IPV4 YANG module for SONiC OS";
+
+    revision 2023-07-19 {
+        description "Initial version";
+    }
+
+    container sonic-dhcp-server-ipv4 {
+        container DHCP_SERVER_IPV4 {
+
+            description "DHCP_SERVER_IPV4 part of config_db.json";
+
+            list DHCP_SERVER_IPV4_LIST {
+
+                description "DHCP_SERVER_IPV4 list part of config_db.json";
+
+                key "name";
+
+                leaf name {
+                    description "Interface name for DHCP server";
+                    type leafref {
+                        path "/vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:name";
+                    }
+                }
+
+                leaf gateway {
+                    description "Gateway IP for DHCP server";
+                    mandatory true;
+                    type inet:ipv4-address;
+                }
+
+                leaf lease_time {
+                    description "Lease time of DHCP IP in second";
+                    mandatory true;
+                    type uint32 {
+                        range "1..4294967295";
+                    }
+                }
+
+                leaf mode {
+                    description "Mode of assigning IP";
+                    mandatory true;
+                    type string {
+                        pattern "PORT" {
+                            error-message "Invalid mode for DHCP server";
+                            error-app-tag dhcp-server-mode-invalid;
+                        }
+                    }
+                }
+
+                leaf netmask {
+                    description "Subnet mask value for DHCP server";
+                    mandatory true;
+                    type inet:ipv4-address-no-zone;
+                }
+
+                leaf-list customized_options {
+                    description "Customized options list";
+                    type leafref {
+                        path "/dhcp-server-ipv4:sonic-dhcp-server-ipv4/dhcp-server-ipv4:DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS/dhcp-server-ipv4:DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS_LIST/dhcp-server-ipv4:name";
+                    }
+                }
+
+                leaf state {
+                    description "State of DHCP server";
+                    mandatory true;
+                    type enumeration {
```

Comment:
https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-templates/sonic-types.yang.j2#L127
How about this type?

Reply:
Fixed